tests/scalers/azure-keyvault-queue.test.ts

```diff
@@ -0,0 +1,182 @@
+})
+
+test.serial(
+  'Deployment should scale with messages on storage defined through trigger auth',
+  async t => {
+    const queueSvc = azure.createQueueService(connectionString)
+    queueSvc.messageEncoder = new azure.QueueMessageEncoder.TextBase64QueueMessageEncoder()
+    await async.mapLimit(
+      Array(1000).keys(),
+      20,
+      (n, cb) => queueSvc.createMessage(queueName, `test ${n}`, cb)
+    )
+
+    // Scaling out when messages available
+    t.true(await waitForDeploymentReplicaCount(1, 'test-deployment', testNamespace, 60, 1000), 'replica count should be 1 after 1 minute')
+
+    queueSvc.clearMessages(queueName, _ => {})
+
+    // Scaling in when no available messages
+    t.true(await waitForDeploymentReplicaCount(0, 'test-deployment', testNamespace, 300, 1000), 'replica count should be 0 after 5 minute')
+  }
+)
+
+test.after.always.cb('clean up azure-queue deployment', t => {
+  const resources = [
+    'scaledobject.keda.sh/test-scaledobject',
+    'triggerauthentications.keda.sh/azure-queue-auth',
+    'secret/test-auth-secrets',
+    'deployment.apps/test-deployment',
+  ]
+
+  for (const resource of resources) {
+    sh.exec(`kubectl delete ${resource} --namespace ${testNamespace}`)
+  }
+  sh.exec(`kubectl delete namespace ${testNamespace}`)
+
+  // delete test queue
+  const queueSvc = azure.createQueueService(connectionString)
+  queueSvc.deleteQueueIfExists(queueName, err => {
+    t.falsy(err, 'should delete test queue successfully')
+    t.end()
+  })
+})
+
+const deployYaml = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+  labels:
+    app: test-deployment
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: test-deployment
+  template:
+    metadata:
+      name:
+      namespace:
+      labels:
+        app: test-deployment
+    spec:
+      containers:
+      - name: test-deployment
+        image: docker.io/kedacore/tests-azure-queue:824031e
+        resources:
+        ports:
+        env:
+        - name: FUNCTIONS_WORKER_RUNTIME
+          value: node
+        - name: AzureWebJobsStorage
+          valueFrom:
+            secretKeyRef:
+              name: test-auth-secrets
+              key: connectionString
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: test-auth-secrets
+  labels:
+data:
+  connectionString: {{CONNECTION_STRING_BASE64}}
+  clientSecret: {{CLIENT_SECRET_BASE64}}
+---
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: azure-keyvault-auth
+spec:
+  azureKeyVault:
+    vaultUri: ${keyvaultURI}
+    credentials:
+      clientId: ${azureADClientID}
+      tenantId: ${azureADTenantID}
+      clientSecret:
+        valueFrom:
+          secretKeyRef:
+            name: test-auth-secrets
+            key: clientSecret
+    secrets:
+    - parameter: connection
+      name: storageConnectionString
```

Comment:
Please use `E2E-Storage-ConnectionString` instead as secret name
```suggestion
      name: E2E-Storage-ConnectionString
```